docs/source/torch.compiler_deepdive.rst

```diff
@@ -206,3 +206,53 @@ By generating this `resume_at` function, we force the remainder of the
 function to be executed in a new Python frame which recursively
 triggers TorchDynamo to restart its capture once execution reaches that
 point for the first time.
+
+How to inspect artifacts generated by TorchDynamo?
+----------------------------------------------
+
+To inspect the artifacts generated by TorchDynamo, there is an API `torch._dynamo.eval_frame._debug_get_cache_entry_list` that retrieves compiled code and guards out of a function's `__code__` object. A compiled function can have several cache entries, and each cache entry consists a generated function to check guards, and a `types.CodeType` object to keep the code to be executed if the guarding conditions are satisfied.
+
+.. code-block:: python
+
+   from torch._dynamo.eval_frame import _debug_get_cache_entry_list
```

Comment:
so what is interesting about the printed output here? or like what does it add on top of the printed assembly code from the section above

Reply:
The section above prints all the compiled code, which is intimidating and hard to understand for large functions (like a whole neural network). This API lets user to extract compiled code for **each individual function**, which is more clear and gives the user more control.